compiler/resolution/visibleFunctions.cpp

```diff
@@ -787,19 +780,20 @@ static void getVisibleFunctionsImpl(const char*       name,
     BlockStmt* next  = getVisibilityScopeNoParentModule(block);
 
     // Recurse in the enclosing block
-    getVisibleFunctionsImpl(name, call, next, visited, nextScopes,
+    getVisibleFunctionsImpl(name, call, next, visInfo, visited,
                             visibleFns, inUseChain);
   }
 
   // Also look at the instantiation point
   if (instantiationPt != NULL)
   {
     INT_ASSERT(!inUseChain);
-    if (nextScopes == NULL)
+    if (visInfo == NULL)
       getVisibleFunctionsImpl(name, call, // visit all POIs right away
-                  instantiationPt, visited, NULL, visibleFns, inUseChain);
+                  instantiationPt, NULL, visited, visibleFns, inUseChain);
     else
-      nextScopes->push_back(instantiationPt); // come back to it later
+      // Overwrites instantiationPt from an outer scope, if any
+      visInfo->nextPOI = instantiationPt; // come back to it later
```

Comment:
I would think this overwrites the innermost scope's POI with the outermost scope's POI? Am I wrong about that for some reason?